### ISSUE TITLE
Navigation

### ISSUE THREAD

**Issue by fikamate** [annikalindberg/react-portfolio#5]
### Responsive Display for Keyboard Users
Description: The navigation menu's visibility is controlled by CSS for different screen sizes, which could potentially hide content from keyboard users.

### Target Area:

const NavItems = styled.ul`
  display: ${props => (props.isMenuOpen ? 'flex' : 'none')};
`;
Solution:
Ensure that content is accessible by not using display: none; to hide it from keyboard users. Instead, use off-screen positioning:

const NavItems = styled.ul`
  // ... other styles
  ${props => !props.isMenuOpen && `
    position: absolute;
    left: -9999px;
    overflow: hidden;
  `}
`;

### Keyboard Navigation for Custom Controls
Description: Custom controls need to be operable by the keyboard, handling both Enter and Space keypresses.

### Target Area:

jsx
Copy code
const handleKeyPress = (e) => {
  if (e.key === 'Enter') {
    onClick(section);
  }
};
### Solution:

Handle Space key along with Enter key:

const handleKeyPress = (e) => {
  if (e.key === 'Enter' || e.key === ' ') {
    onClick(section);
  }
};

<nav aria-label="Main navigation">
    
</nav>
Proper ARIA Labels: If you're using a <div> or <ul> for navigation and cannot change it to a <nav>, you can use ARIA roles to define it as a navigation section.

<div role="navigation" aria-label="Main navigation">
    <ul>
        
    </ul>
</div>
Correct Use of aria-label or aria-labelledby: If you have multiple navigation sections, use aria-label or aria-labelledby to provide unique descriptions for each.

- [x]  For List Items in Navigation:

Descriptive Text for Links: Each list item (<li>) should contain an anchor (<a>) or a button with descriptive text that VoiceOver can read. If the list items are empty or contain only an icon (like an SVG), you need to add text or aria-label so that VoiceOver can announce what the link or button is for.

<li>
    <a href="/home" aria-label="Home">Home</a>
</li>

- [x] SVGs as Icons in Links: If you're using SVGs as icons inside your list items, make sure to include descriptive title tags within the SVG or use aria-hidden="true" on the SVG and provide an accessible name for the link or button containing the SVG.
- [ ] 
<li>
    <a href="/home" aria-label="Home">
        <svg aria-hidden="true">...</svg>
        <span class="visually-hidden">Home</span>
    </a>
</li>

The menu seams to have list items (<li>) with tabindex="0" and aria-expanded attributes, suggesting these are meant to be interactive elements, possibly expandable/collapsible. If no proper text or labels, VoiceOver cannot provide meaningful announcements for these elements.

The aria-expanded attribute also indicates the presence of a dropdown or expandable menu. Make sure that when such an element is focused, VoiceOver correctly announces whether it is expanded or collapsed.

To summarize, if VoiceOver is only announcing "group" and not that it is a navigation element, you likely need to:

Ensure that navigation elements are wrapped in a <nav> or have the correct role attribute.
Provide descriptive names for links, especially when using icons or SVGs.
Use aria-label to describe the purpose of each navigation item, particularly if the text is not visually displayed.
